cubedash/_pages.py

```diff
@@ -178,18 +178,39 @@ def search_page(
     if product_name:
         query["product"] = product_name
 
-    if "time" in query:
-        # If they left one end of the range open, fill it in with the product bounds.
+    if "dataset_maturity" in query:
+        query["dataset_maturity"] = query["dataset_maturity"].lower()
+
+    # The URL time range always trumps args.
+    if time_range:
+        query["time"] = time_range
+    elif "time" in query:
         search_time = query["time"]
-        assert isinstance(search_time, Range)
+        # If it's not a range, it's almost certainly because we're searching via an
+        # individual dataset field value, so we need to create a range corresponding to that day.
+        if not isinstance(search_time, Range):
+            search_time = Range(search_time, search_time + timedelta(days=1))
+        # If they left one end of the range open, fill it in with the product bounds.
         if product_summary:
-            query["time"] = Range(
+            search_time = Range(
                 search_time.begin or product_summary.time_earliest,
                 search_time.end or product_summary.time_latest + timedelta(days=1),
             )
-    # The URL time range always trumps args.
-    if time_range:
-        query["time"] = time_range
+        query["time"] = search_time
+
+    # same logic as with 'time'
+    if "creation_time" in query:
```

Comment:
does this include? https://github.com/opendatacube/datacube-core/issues/1295

Reply:
do you think you can set another integration test for this query?based off the below test?

https://github.com/opendatacube/datacube-explorer/blob/develop/integration_tests/test_page_loads.py#L515-L522

Reply:
Done!